app/src/main/java/com/example/mytestproject/ui/weatherData/WeatherDataFragment.kt

```diff
@@ -39,12 +42,35 @@ class WeatherDataFragment : Fragment() {
         animationDrawable.start()
 
         setupWeatherData()
+
+        btnRetry.setOnClickListener {
+            weatherDataViewModel.getWeather()
+            setupWeatherData()
+        }
+
     }
 
     private fun setupWeatherData() {
-        weatherDataViewModel.weatherDataApi.observe(viewLifecycleOwner, Observer {
-            weatherDataBinding.weatherData = it
+        weatherDataViewModel.viewState.observe(viewLifecycleOwner, Observer {
+            when (it) {
+                WeatherViewState.Loading -> {
+                    group_temp_abbreviation.visibility = View.GONE
```

Comment:
group_error_views is also should be hidden while loading. Otherwise, after error, user will see retry button while data is reloading.